src/main/java/pl/tomaszdziurko/jvm_bloggers/rest/blogpost_redirect/RedirectController.java

```diff
@@ -10,27 +16,59 @@
 
 import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost;
 import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPostRepository;
+import pl.tomaszdziurko.jvm_bloggers.click_counter.ClicksStoringActor;
+import pl.tomaszdziurko.jvm_bloggers.click_counter.SingleClick;
+import pl.tomaszdziurko.jvm_bloggers.click_counter.domain.ClickRepository;
+import pl.tomaszdziurko.jvm_bloggers.utils.NowProvider;
 import pl.tomaszdziurko.jvm_bloggers.utils.UriUtmComponentsBuilder;
 
 import java.io.IOException;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
 import javax.servlet.http.HttpServletResponse;
 
+import static akka.actor.ActorRef.noSender;
+
 @RestController
-@RequestMapping(path = "/r")
-@AllArgsConstructor(onConstructor = @__(@Autowired))
+@RequestMapping(path = RedirectController.REDIRECT_URL_PATH)
+@Slf4j
 public class RedirectController {
 
     private static final String UTM_MEDIUM = "link";
+    private static final int MAX_ALLOWED_EXECUTION_TIME_IN_MILLIS = 1000;
+    public static final String REDIRECT_URL_PATH = "/r";
 
     private final BlogPostRepository blogPostRepository;
+    private final ActorRef actorRef;
+
+    @Autowired
+    public RedirectController(BlogPostRepository blogPostRepository,
+                              ClickRepository clickRepository,
+                              ActorSystem actorSystem,
+                              NowProvider nowProvider) {
+        this.blogPostRepository = blogPostRepository;
+        this.actorRef = actorSystem.actorOf(
+            new RoundRobinPool(3).props(ClicksStoringActor.props(clickRepository, nowProvider)),
+            "clicksStoringActor"
+        );
+    }
 
     @RequestMapping(value = "/{uid}", method = RequestMethod.GET)
     public void redirectToBlogPostWithUid(HttpServletResponse response, @PathVariable String uid) {
+        Stopwatch stopwatch = Stopwatch.createStarted();
         Optional<BlogPost> blogPost = blogPostRepository.findByUid(uid);
         if (blogPost.isPresent()) {
+            actorRef.tell(new SingleClick(blogPost.get()), noSender());
             redirectToBlogPost(response, blogPost.get());
+            long executionTime = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+            if (executionTime > MAX_ALLOWED_EXECUTION_TIME_IN_MILLIS) {
+                log.warn(
```

Comment:
What is the business purpose of logging redirect execution time? Why 1000ms is a critical value for that?